Incorrect @octokit TypeScript typings. `meta` property in Link type should be replaced by `headers`.

**Description:**
I faced with this error during building of TypeScript project:

> error TS2345: Argument of type 'Response<any>' is not assignable to parameter of type 'Link'.
>   Type 'Response<any>' is not assignable to type '{ meta: { link: string; }; }'.
>     Property 'meta' is missing in type 'Response<any>'. 

**Environment:**
- TypeScript: 2.9.2
- Node: 8.11.2
- @octokit/rest: 15.11.0

**Root cause:**
Templates (`index.d.ts.tpl` and `index.js.flow.tpl`) which are used to create typigns are incorrect. For `Link` type it contains deprecated `meta` property which was replaced with `headers`.

Incorrect Link type:
```
export type Link =
    | { link: string; }
    | { meta: { link: string; }; }
    | string;
```

**Fix:** Update templates to replace `meta` with `headers`;

--- a/scripts/templates/index.d.ts.tpl
+++ b/scripts/templates/index.d.ts.tpl
@@ -120,7 +120,7 @@ declare namespace Github {
 
   export type Link =
     | { link: string; }
-    | { meta: { link: string; }; }
+    | { headers: { link: string; }; }
     | string;
 
   export interface Callback<T> {

--- a/scripts/templates/index.js.flow.tpl
+++ b/scripts/templates/index.js.flow.tpl
@@ -61,7 +61,7 @@ declare module "github" {
 
   declare type Link =
     | { link: string; }
-    | { meta: { link: string; }; }
+    | { headers: { link: string; }; }
     | string
     | any;
 

